docs/admin/access.rst

```diff
@@ -1,333 +1,439 @@
+        "Role" -- "Permission";
+        "Group" -- "Project";
+        "Group" -- "Language";
+        "Group" -- "Components";
+        "Group" -- "Component list";
+    }
 
-Weblate comes with a predefined set of groups for a project, wherefrom you can assign
-users.
+.. note::
 
+  A group can have no roles assigned to it, in that case access to browse the
+  project by anyone is assumed (see below).
 
-.. describe:: Translate
+Access for browse to a project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    Can translate the project, and upload translations made offline.
+A user has to be a member of a group linked to the project, or any component
+inside that project. Having membership is enough, no specific permissions are
+needed to browse the project (this is used in the default `Viewers` group, see
+:ref:`default-groups`).
 
-.. describe:: Sources
+Access for browse to a component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    Can edit source strings in :ref:`monolingual` and source string info.
+A user can access unrestricted components once able to access the components'
+project (and will have all the permissions the user was granted for the
+project). With :ref:`component-restricted` turned on, access to the component
+requires explicit permissions for the component (or a component list the component is in).
 
-.. describe:: Languages
+.. _perm-check:
 
-    Can manage translated languages (add or remove translations).
+Scope of groups
+^^^^^^^^^^^^^^^
 
-.. describe:: Glossary
+The scope of the permission assigned by the roles in the groups are applied by
+the following rules:
 
-    Can manage glossary (add or remove entries, or upload).
+- If the group specifies any :guilabel:`Component list` all the permissions given to
+  members of that group are granted for all the components in the component
+  lists attached to the group and an access with no additional permissions is
+  granted for all the projects these components are in. :guilabel:`Components`
+  and :guilabel:`Projects` are ignored.
 
-.. describe:: Memory
+- If the group specifies any :guilabel:`Components` all the permissions given to
+  the members of that group are granted for all the components attached to the
+  group and an access with no additional permissions is granted for all the
+  projects these components are in. :guilabel:`Projects` are ignored.
 
-    Can manage translation memory.
+- Otherwise, if the group specifies any :guilabel:`Projects` either by directly
+  listing them or by having :guilabel:`Selected projects` set to a value like `All
+  public`, all those permissions are applied to all the projects, which
+  effectively grants the same permissions to access all projects'
+  :ref:`unrestricted components <component-restricted>`.
 
-.. describe:: Screenshots
+- The restrictions imposed by a group's :guilabel:`Languages` are applied later,
+  when the access to perform a certain action is verified, and it only affects
+  some translation-related permissions.
```

Comment:
```suggestion
- The restrictions imposed by a group's :guilabel:`Languages` are applied separately,
  when it is verified if a user has an access to perform certain actions. Namely,
  It's applied only to actions directly related to translation process itself like
  reviewing, saving translations, adding suggestions, etc.
```

I want to rephrase the paragraph, but not sure. What do you think?

Reply:
Longer for dubious benefit. Let's just get the formatting and info in check and I can come in and deal with the rephrasing in a separate PR.

Reply:
@Fat-Zer 
> I want to rephrase the paragraph, but not sure. What do you think?

I think it’s good.